firmware/Sources/Test/HAL/hal_sts_test.c

```diff
@@ -0,0 +1,70 @@
+/*                              Include other headers                             */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                     Definition of local symbolic constants                     */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                    Definition of local function like macros                    */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*            Definition of local types (typedef, enum, struct, union)            */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                         Definition of local variables                          */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                        Definition of exported variables                        */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                      Definition of exported constant data                      */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                    Declaration of local function prototypes                    */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                       Definition of local constant data                        */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                         Definition of local functions                          */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                        Definition of exported functions                        */
+/**********************************************************************************/
+
+HAL_STS_result_e HAL_StsTest(void){
+	HAL_STS_result_e res = HAL_STS_RESULT_ERROR;
+	uint16_t temp;
+	uint8_t i=0;
+	while (i<5){
+		res = HAL_StsReadTemperature(&temp);
+		if (res == HAL_STS_RESULT_ERROR) break;
```

Comment:
never use breaks in c. ONLY ONE EXIT POINT

Reply:
Changed in last commit